src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -111,7 +111,13 @@ public Point TransformOrigin
 
         public bool ViewingHomespace
         {
-            get { return model.CurrentWorkspace == HomeSpace; }
+            get {
+                if (model.CurrentWorkspace != HomeSpace)
+                {
+                    onNotViewingHomeSpace();
```

Comment:
Making calls or throwing events in getter usually is not desired. We sometimes do it in setter but since in Dynamo we already have OnPropertyChanged() function ready for this, I would move this to there, there is a switch block to handle property change and what would be the consequences. We just need to flip the visibility based on the value of this property I think.

Reply:
Done